File: archive-application/src/main/java/site/archive/dto/v1/user/SpecificUserDtoV1.java
package site.archive.dto.v1.user;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import site.archive.common.DateTimeUtils;
import site.archive.domain.user.BaseUser;
import site.archive.domain.user.UserRole;

@RequiredArgsConstructor
@Getter
public class SpecificUserDtoV1 {

    private final Long userId;
    private final String mailAddress;
    private final UserRole userRole;
    private final String createdAt;
    private final String profileImage;
    private final String nickname;
    private final String userType;

    public static SpecificUserDtoV1 from(BaseUser baseUser) {
        var createdAt = DateTimeUtils.getDateTimeFormatter().format(baseUser.getCreatedAt());
        return new SpecificUserDtoV1(baseUser.getId(),
                                     baseUser.getMailAddress(),
                                     baseUser.getRole(),
                                     createdAt,
                                     baseUser.getProfileImage(),
                                     baseUser.getNickname(),
                                     baseUser.getUserType());
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v1/user/SpecificUserDtoV1.kt
package site.archive.dto.v1.user

import site.archive.common.dateTimeFormatter
import site.archive.domain.user.BaseUser
import site.archive.domain.user.UserRole

data class SpecificUserDtoV1(
    val userId: Long?,
    val mailAddress: String,
    val userRole: UserRole,
    val createdAt: String,
    val profileImage: String,
    val nickname: String,
    val userType: String
) {

    companion object {
        @JvmStatic
        fun from(baseUser: BaseUser) : SpecificUserDtoV1 {
            val createdAt = dateTimeFormatter.format(baseUser.createdAt)
            return SpecificUserDtoV1(
                userId = baseUser.id,
                mailAddress = baseUser.mailAddress,
                userRole = baseUser.role,
                createdAt = createdAt,
                profileImage = baseUser.profileImage,
                nickname = baseUser.nickname,
                userType = baseUser.userType
            )
        }
    }

}